core/base-service/cache-headers.js

```diff
@@ -39,14 +39,14 @@ function coalesceCacheLength({
   assert(defaultCacheLengthSeconds !== undefined)
 
   const cacheLength = coalesce(
+    serviceOverrideCacheLengthSeconds,
     serviceDefaultCacheLengthSeconds,
     defaultCacheLengthSeconds
   )
 
   // Overrides can apply _more_ caching, but not less. Query param overriding
   // can request more overriding than service override, but not less.
   const candidateOverrides = [
-    serviceOverrideCacheLengthSeconds,
     overrideCacheLengthFromQueryParams(queryParams),
```

Comment:
For the history on this, see https://github.com/badges/shields/pull/2755

Basically, the only thing relying on the logic as it stood here was the endpoint badge (we don't want endpoint badge users to be able to set a lower `cacheSeconds` value than the service default). I've moved this logic so it gets applied in `endpoint.service.js` so that we can now lower `cacheSeconds` with a custom exception property if we want. If we left this as it was, we'd only be able to raise `cacheSeconds` with a custom exception property.